gpu-alloc/src/allocator.rs

```diff
@@ -167,27 +167,32 @@ where
             return Err(AllocationError::NoCompatibleMemoryTypes);
         }
 
-        let strategy = strategy.unwrap_or_else(|| match request.dedicated {
-            Dedicated::Required => Strategy::Dedicated,
-            Dedicated::Preferred
-                if request.size >= self.preferred_dedicated_treshold
-                    && self.allocations_remains > 0 =>
-            {
-                Strategy::Dedicated
-            }
-            _ if request.usage.contains(UsageFlags::TRANSIENT) => {
-                if request.size > self.transient_dedicated_treshold && self.allocations_remains > 0
+        let strategy = match strategy {
+            Some(Strategy::Linear) if request.size > self.linear_chunk => Strategy::Dedicated,
```

Comment:
You are right that this becomes less of a strong order. However, the library currently crashes on this scenario, which is less desired. I think falling back from either linear or buddy allocation to the dedicated can be seen as an implementation detail. From the user point of view, they behave like the the other linear and buddy allocated blocks.

To clarify: the main scenario that an explicit strategy saves from is when user thinks it's Buddy, but in reality it turns out Linear, and it lives for much longer than anticipated. Same for the other direction. These concerns aren't applicable to Dedicated as a fallback to any of these.

Reply:
`Buddy` block can be deallocated quickly with no problems if user thought it is `Linear`.

How about fallback from `Linear` to `Buddy` if configured that `Buddy` can serve requested size and `Linear` can't?

I still believe that using `UsageFlags::TRANSIENT` to specify that memory block will not live for long is better than relying on internal knowledge how `Linear` and `Buddy` sub-allocators work to decide on which to use.

Reply:
Is it guaranteed that TRANSIENT picks the linear allocator?

Reply:
It is guaranteed that `TRANSIENT` picks best suited sub-allocator for the request considering other properties.
Currently it picks `Linear` allocator unless requested size is greater than or equal to treshold in which case `Dedicated` allocator is used.

Reply:
If we stop using the `Strategy`, I'm also worried about the block of code (affected in this PR) that tries to guess whether to use dedicated allocation. It has a ton of constants with unclear values. I don't want to think about it, I just want a linear allocation when possible in some cases.

Reply:
There are 3 constants involved, all to pick `Dedicated` strategy instead of `Linear` or `Buddy`.

I agree that `transient_dedicated_treshold` is somewhat unclear. It supposed to make allocator pick `Dedicated` instead of using almost whole chunk of linear allocator.

Reply:
There are 3 constants involved, and neither of them is `linear_chunk`, which I'd expect to be the only thing involved, like in this PR.

Reply:
Because allocating `linear_chunk - N` bytes in `LinearAllocator` is meaningless for small `N`.
That's because `transient_dedicated_treshold` must be merely bound by `linear_chunk` (erroneously this code is missing).
`dedicated_treshold` is smaller value for non-transient usage. And `preffered_dedicated_treshold` is for allocations that would prefer dedicated memory (intended to be bound to resources that prefer dedicated memory)

Reply:
It's meaningless but also totally harmless. The amount of wasted space is supposedly negligible.
Could we remove `transient_dedicated_treshold` altogether with this in mind?

Reply:
In this case you can simply make `transient_dedicated_treshold = linear_chunk` and call it a day.
Later you will be able to revisit this and make another decision.